Tasks/DotnetCLIRestore/restorecommand.ts

```diff
@@ -0,0 +1,270 @@
+        let externalAuth = tl.getEndpointAuthorization(endpointName, true);
+        let scheme = tl.getEndpointAuthorizationScheme(endpointName, true).toLowerCase();
+        let token = "";
+        switch(scheme) {
+            case "token":
+                token = externalAuth.parameters["AccessToken"];
+                tl.debug("adding token auth entry for feed GitHub");
+                externalAuthArray.push(new auth.TokenExternalAuthInfo(<auth.IPackageSource>
+                    {
+                        feedName: "github",
+                        feedUri: "https://nuget.pkg.github.com/" + username + "/index.json"
+                    },
+                    token));
+                break;
+            case "personalaccesstoken":
+                token = externalAuth.parameters["accessToken"];                
+                tl.debug("adding token auth entry for feed GitHub");
+                externalAuthArray.push(new auth.TokenExternalAuthInfo(<auth.IPackageSource>
+                    {
+                        feedName: "github",
+                        feedUri: "https://nuget.pkg.github.com/" + username + "/index.json"
+                    },
+                    token));
+                break;
+            case "usernamepassword":
+            case "none":
+                break;
+            default:
+                break;
+        }
+    });
+
+    return externalAuthArray;
+}
+
+function GetGitHubUser(endpointId: string): Promise<string> {
+    let externalAuth = tl.getEndpointAuthorization(endpointId, true);
+    let scheme = tl.getEndpointAuthorizationScheme(endpointId, true).toLowerCase();
+
+    if (!(scheme == "token" || scheme == "personalaccesstoken")) {
+        return new Promise((resolve, reject) => {
+            resolve("");
+        });
+    }
+
+    let token = "";
+    if (scheme == "token") {
+        token = externalAuth.parameters["AccessToken"];
+    } else if (scheme == "personalaccesstoken") {
+        token = externalAuth.parameters["accessToken"];
+    }
+
+    var url = "https://api.github.com/user";
+
+    return new Promise((resolve, reject) => {
+        request.get({
+            url : url,
+            headers : {
+                "Authorization": "Token " + token,
+                "User-Agent": "azure-pipelines"
+            }
+        }, function(error, response, body) {
+            if (error) reject(error);
+            let responseJson = JSON.parse(body);
+            resolve(responseJson["login"]);
+        });
+    });
+}
+
+function dotNetRestoreAsync(dotnetPath: string, projectFile: string, packagesDirectory: string, version: string, configFile: string, noCache: boolean, verbosity: string): Q.Promise<number> {
+    const dotnet = tl.tool(dotnetPath);
+    dotnet.arg('add');
+
+    if (projectFile) {
+        dotnet.arg(projectFile);
+    }
+
+    if (packagesDirectory) {
+        dotnet.arg('package');
+        dotnet.arg('OctocatApp');
```

Comment:
will it always be octocatapp ?

Reply:
This is a test code that crept in. Removed it.